samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlConfigRewriter.java

```diff
@@ -0,0 +1,84 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql.runner;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.samza.config.Config;
+import org.apache.samza.config.ConfigRewriter;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.sql.interfaces.SourceResolver;
+import org.apache.samza.sql.interfaces.SystemStreamConfig;
+import org.apache.samza.sql.testutil.SamzaSqlQueryParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Samza SQL config rewriter is used to add few configs that are derived from the existing configs.
+ * The configs that are added to the existing set are
+ *  CFG_SQL_STMTS_JSON - Json serialized collection of SQL statements read from the SQL file
+ *  streams.{streamName}.samza.system - System corresponding to the input and output streams.
+ */
+public class SamzaSqlConfigRewriter implements ConfigRewriter {
```

Comment:
Please have the logic inside your sql runner instead of using a rewriter.

Reply:
Yup i will move this to SamzaSqlApplicationRunner and avoid using the ConfigRewriter.